Fixing inconsistencies

tognee opened the pull request:
## What does this do?
Adds missing blank objects where they should be

## How was it tested?
Parsed with a JSON parser and tested with a python script

fergcb commented:
Starting equipment fix looks good.

Can't say I'm a big fan of all the empty objects, though. I'd personally prefer these fields to be optional. With the empty objects there, the presence of the object suggests the value exists, but on inspection, the value would be invalid.

Perhaps splitting these two unrelated proposals into different PRs would be a good idea, so the fix can go through while we discuss the empty objects proposal?

tognee commented:
Moved the paladin fix to #324 

I've added the blank objects as there were already blank objects in just some properties, but not all.
This was brought to my attention by my python script.

The solution should be adding the empty objects (as this pr does) or remove them all, while keeping only the ones that matter

fergcb commented:
My vote would be to axe the empties, for sure. An empty object doesn't sufficiently represent the absence of an optional field, imo.

tognee commented:
Take a look at all the files, here are my notes:
Didn't check Monsters and Levels empty arrays, as I can't rn
Let me know your thoughts

Empty arrays

Ability-Scores
"con" -> "skills"
It should stay as it has no skills, unlike the other abilities

Equipment
"flail", "morningstar" and "war-pick" have "properties" as an empty Array
It should stay as all the other equipments have properties

Features
11 Features have "prerequisites" as an Empty Array
It should be removed as other Features without prerequisites don't have an empty Array

Monsters
1122 empty Arrays

Levels
334 empty Arrays

Proficiencies
All empty Arrays refer to classes and races
Should stay

Races
"starting_proficiencies", "traits" and "subraces" have empty Arrays
IMO should stay if empty

Spells
All refer to "subclasses"
Should stay

StartingEquipment
empty starting_equipment in "fighter"
Should stay

Subraces
"starting_proficiencies" and "languages" have empty Arrays
Should stay

Traits
"races", "subraces" and "proficiencies"
Should stay

Empty objects

Features
"subclasses" have empty objects
Should stay

Levels
"subclass" should stay
empty "spellcasting" should be removed on all classes and subclasses that can't use spells
Maybe removing from subclasses as well?

Races
"ability_bonus_options", "starting_proficiency_options", "language_options", "trait_options"
IMO should stay if empty

Spells
"heal_at_slot_level" in "greater-restoration"
Not sure

Subraces
"ability_bonus_options", "starting_proficiency_options", "language_options", "racial_trait_options"
IMO should stay if empty

fergcb commented:
Empty arrays should stay, absolutely. I only recommended that empty *objects* should be cut.

I don't think *any* of the attributes you listed with object values should stay if empty. Ones ending in `_option`, for example, represent choices. If there is no choice to be made, the attribute probably shouldn't be there, rather than being an empty object (which looks at first glance like something should be there, but doesn't actually follow the schema of an Choice object).

bagelbits commented on package-lock.json:
```diff
@@ -0,0 +1,6520 @@
+{
```
Currently, we're using yarn for this repo, which is why there is no `package-lock.json`. Can you remove this for now?

bagelbits commented:
I think I still see some empty arrays that you removed which should be put back?

tognee commented:
I removed "prerequisites" in Features as not all had that array
Should I put back only the ones I removed or also add the missing ones?

ogregoire commented:
This seems good. Given the various forths and backs, I suggest we squash the commits.

bagelbits commented:
I believe you'll want to rebase.
